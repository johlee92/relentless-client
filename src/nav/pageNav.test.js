import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PageNav from './pageNav';
import { BrowserRouter } from 'react-router-dom';

describe(`PageNav component`, () => {
    const props = {
        className: 'test-class-name'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <PageNav />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component by default', () => {
        const wrapper = shallow(<PageNav />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the component from props', () => {
        const wrapper = shallow(<PageNav {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
