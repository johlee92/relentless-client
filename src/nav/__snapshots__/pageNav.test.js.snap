// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNav component renders the component by default 1`] = `
<nav
  className="PageNav"
>
  <section>
    <Link
      to="/"
    >
      Relentless
    </Link>
  </section>
  <section>
    <Link
      to="/user1"
    >
      User1
    </Link>
  </section>
</nav>
`;

exports[`PageNav component renders the component from props 1`] = `
<nav
  className="PageNav"
>
  <section>
    <Link
      to="/"
    >
      Relentless
    </Link>
  </section>
  <section>
    <Link
      to="/user1"
    >
      User1
    </Link>
  </section>
</nav>
`;
